chain-api/src/types/LaunchPadSale.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  BigNumberProperty,
+  ChainKey,
+  ChainObject,
+  StringEnumProperty,
+  TokenInstanceKey
+} from "@gala-chain/api";
+import BigNumber from "bignumber.js";
+import { Exclude } from "class-transformer";
+import { IsNotEmpty, IsString } from "class-validator";
+
+export const INITIAL_LAUNCHPAD_FEE_ADDRESS = "client|ops-admin";
+export const MARKET_CAP = "1640985.8441726";
+export const BASE_PRICE = "16506671506650";
+
+export enum SaleStatus {
+  ONGOING = "Ongoing",
+  END = "Finished"
+}
+
+export class LaunchPadSale extends ChainObject {
+  @Exclude()
+  static INDEX_KEY = "LAUNCHPAD";
+
+  @ChainKey({ position: 0 })
+  @IsString()
+  @IsNotEmpty()
+  public vaultAddress: string;
+
+  @StringEnumProperty(SaleStatus)
+  public SaleStatus: SaleStatus;
+
+  public sellingToken: TokenInstanceKey;
+  public nativeToken: TokenInstanceKey;
+  public sellingTokenQuantity: string;
+  public nativeTokenQuantity: string;
+
+  public saleOwner: string;
```

Comment:
Every property in the on-chain data model that extends the base `ChainObject` class must have at least one decorator from the `class-validator` library.

```suggestion
  @ValidateNested()
  @Type(() => TokenInstanceKey)
  public sellingToken: TokenInstanceKey;
  
  @ValidateNested()
  @Type(() => TokenInstanceKey)
  public nativeToken: TokenInstanceKey;
  
  @IsNotEmpty()
  public sellingTokenQuantity: string;
  
  @IsNotEmpty()
  public nativeTokenQuantity: string;

  @IsNotEmpty()
  public saleOwner: string;
```